core-bundle/src/Image/Studio/FigureBuilder.php

```diff
@@ -597,14 +603,19 @@ private function onDefineMetadata(): ?Metadata
         $metadata = $this->filesModel->getMetadata(...$locales);
 
         if (null !== $metadata) {
-            return $metadata;
+            return $metadata->with($fileReferenceData);
         }
 
         // If no metadata can be obtained from the model, we create a
         // container from the default meta fields with empty values instead
         $metaFields = $this->filesModelAdapter()->getMetaFields();
 
-        return new Metadata(array_combine($metaFields, array_fill(0, \count($metaFields), '')));
+        $data = array_merge(
+            array_combine($metaFields, array_fill(0, \count($metaFields), '')),
+            $fileReferenceData
+        );
+
+        return new Metadata($data);
```

Comment:
```suggestion
        $metadata = new Metadata(array_combine($metaFields, array_fill(0, \count($metaFields), '')));

        return $metadata->width($fileReferenceData);
```

May be? Then the merging is consistently handled via `Metadata::with`.

Reply:
Then you'd potentially create two instances vs one though. :fox_face:

Reply:
True, but that's the case with the previous returns anyway, isn't it?

Reply:
The other two cases aren't build from scratch but from an explicitly set metadata object, though. There we need to go this route whereas in this case we're constructing the set on the fly.

I mean, at the end it's what you prefer, the memory + performance difference is negligible.